creator/data_reviews/mutations.py

```diff
@@ -0,0 +1,323 @@
+
+        # Check files in review
+        review_file_ids = check_review_files(input, study)
+
+        # Create data_review
+        with transaction.atomic():
+            data_review = DataReview(
+                **{k: input[k] for k in input if k not in {"study", "files"}}
+            )
+            data_review.study = study
+            data_review.creator = user
+            data_review.save()
+
+            if review_file_ids:
+                data_review.files.set(review_file_ids)
+                # Start review if we have files
+                data_review.start()
+                data_review.save()
+
+        return CreateDataReviewMutation(data_review=data_review)
+
+
+class UpdateDataReviewMutation(graphene.Mutation):
+    """ Update an existing data_review """
+
+    class Arguments:
+        id = graphene.ID(
+            required=True, description="The ID of the data_review to update"
+        )
+        input = UpdateDataReviewInput(
+            required=True, description="Attributes for the data_review"
+        )
+
+    data_review = graphene.Field(DataReviewNode)
+
+    def mutate(self, info, id, input):
+        """
+        Updates an existing data_review
+        """
+        user = info.context.user
+
+        model, node_id = from_global_id(id)
+        try:
+            data_review = DataReview.objects.get(pk=node_id)
+        except DataReview.DoesNotExist:
+            raise GraphQLError("DataReview was not found")
+
+        # Check permissions
+        # Must have general change review permission for any study OR
+        # permission to change review for user's studies
+        if not (
+            user.has_perm("data_reviews.change_datareview")
+            or (
+                user.has_perm("data_reviews.change_my_study_datareview")
+                and user.studies.filter(kf_id=data_review.study.kf_id).exists()
+            )
+        ):
+            raise GraphQLError("Not allowed")
+
+        # Closed/Completed reviews cannot be modified
+        if data_review.state in {State.CLOSED, State.COMPLETED}:
+            raise GraphQLError(
+                "Cannot modify a data review that has been closed or "
+                "completed. However, a closed review can be re-opened and "
+                "then modified."
+            )
+
+        # Check files in review
+        review_file_ids = check_review_files(input, data_review.study)
+
+        # Update data_review
+        with transaction.atomic():
+            for attr in ["name", "description"]:
+                if attr in input:
+                    setattr(data_review, attr, input[attr])
+
+            if review_file_ids:
+                # Start review if not started
+                if data_review.state == State.NOT_STARTED:
+                    data_review.start()
+
+                # We received updates while waiting, go back to reviewing state
+                elif data_review.state == State.WAITING:
```

Comment:
So if we update the review's name or description while it's in the `waiting` state, it will go back to the `reviewing` state even if the files didn't change? I think we need to check if the new list of files is different from the old list of files in order to make this transition.

Reply:
If the name/description change it has no effect on the state of the review bc this code only executes if "files" are in the input. However, you bring up a good point. I should only update from waiting to in review if the files actually changed. Right now that state is just based on the presence of "files" in the input and whether they're all valid.

Reply:
What if we have a review with state `{"name": "ABC", "files": ["A","B"]}`.
Then we receive an update with payload `{"name": "123", "files": ["A","B"]}`.
Wouldn't this cause it to back to the reviewing state despite the files not changing?

Reply:
No, bc I don't think name/description changes should trigger the data review state to change